### PR TITLE
Add problem-solving skill and problem-solver subagent (Pólya method)

### PR DESCRIPTION
**Pull request by kanade0404**
## Summary

- Adds `skills/problem-solving/`: applies George Pólya's four-step method from *How to Solve It* (理解する / 計画をたてる / 実行する / 振り返る) as a general, domain-agnostic problem-solving heuristic — not limited to coding (ambiguous tasks, unclear bugs, stuck implementations, choosing between approaches). Frontmatter negative space draws explicit boundaries against `design`, `tdd`/`tidy-first`, `grill-with-docs`, and `research-practices` so it doesn't compete with them. The exhaustive guiding-question checklist (transcribed from the source book) lives in `references/polya-checklist.md` per progressive-disclosure convention.
- Adds `subagents/problem-solver.md`: the first real entry in the `subagents/` rulesync feature slot (previously placeholder-only). A canonical rulesync subagent definition for dispatching a single, fully-specified hard problem (clear-repro bug, isolated algorithm/puzzle) to a one-shot agent that runs the same four steps and returns a structured Understand/Plan/Execute/Look-Back report.
- Updates `rules-local/repository-guidelines.md` and `subagents/README.md` to reflect that `subagents/` now has real content (only `commands/`/`hooks/` remain placeholders), and regenerates `CLAUDE.md` / `AGENTS.md` / `.claude/` / `.agents/` via `node scripts/rulesync-sync.mjs`.

## Eval results

- `skills/problem-solving/evals/problem-solving-trigger.json`: 20 cases (7 explicit, 3 ambiguous, 6 adjacent, 4 distractor).
- Scored against `problem-solving-trigger-results-2026-08-08.jsonl`: **precision=1.000, recall=1.000, F1=1.000** (no FP/FN).

## Test plan

- [x] `uv run python3 -m unittest discover -s tests` — 98 tests pass
- [x] `uv run python .github/scripts/check_trigger_evals.py` — exit 0, no mismatches for `problem-solving` (or any other skill)
- [x] `node scripts/rulesync-sync.mjs --check` — up to date (generated outputs committed)
- [x] `uv run python skills/skill-builder/scripts/score_triggers.py --cases skills/problem-solving/evals/problem-solving-trigger.json --preds skills/problem-solving/evals/problem-solving-trigger-results-2026-08-08.jsonl` — F1=1.0

---
_Generated by [Claude Code](https://claude.ai/code/session_016TkNEP9AnFDA257kmZLeNG)_

## Summary by CodeRabbit

* **新機能**
  * Pólyaの4段階法に基づく問題解決スキルを追加しました。
  * 問題の理解、計画、実行、振り返りを段階的に支援します。
  * 複数の計画案と独立した検証結果を比較できるワークフローを追加しました。
  * 単一の問題を処理する問題解決エージェントを追加しました。

* **ドキュメント**
  * 問題解決の自問チェックリストと適用範囲を追加しました。
  * スキルの判定基準と評価ケースを整備しました。
  * 配布機能の構成と検証手順を更新しました。

### CODE REVIEW

**Comment by kanade0404** [comment]
@coderabbitai review

**Comment by kanade0404** [comment]
### conflict 解決完了

- 解決方針: merge (`master` → このブランチ)
- 背景: `master` に #104 (harness/minimal-claude-md-rulesync-skill) 他がマージされ、`rules-local/repository-guidelines.md` の削除・root rule (`orchestration-policy.md`) への縮小・`rulesync-sync` skill の新設という大きな再構成が入ったため、生成物 (`CLAUDE.md`/`AGENTS.md`) と `rules-local/repository-guidelines.md` が衝突。
- 解決したファイル:
  - `CLAUDE.md` / `AGENTS.md` — 生成物のため `master` 側 (`--theirs`) を採用し、後段で `node scripts/rulesync-sync.mjs` により再生成
  - `rules-local/repository-guidelines.md` — `master` 側の削除方針に追従（内容は `skills/rulesync-sync/SKILL.md` に既に統合済みと確認）
  - `skills/rulesync-sync/SKILL.md`（および `.claude/`/`.agents/` ミラー） — このファイルが `repository-guidelines.md` 削除後の新しい正典であることを確認し、`subagents/` に関する記述を「現状 placeholder」から「`problem-solver.md` を含む」旨に更新（本 PR の内容を反映する形でこちらも追従）
- 検証結果:
  - `node scripts/rulesync-sync.mjs --check`: PASS (drift なし)
  - `uv run python3 -m unittest discover -s tests`: PASS (106 tests)
  - `uv run python .github/scripts/check_trigger_evals.py`: PASS (exit 0, mismatch なし)
- 残課題: なし

push: `07954ef`

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by kanade0404** [comment]
Re: the Biome tool-failure notice on `four-step-pipeline.workflow.js` ("Illegal return statement outside of a function", line 205, all 3 copies) —

Not a bug, working as intended. This file isn't a standalone Node module; it's a script for Claude Code's `Workflow` tool, whose runtime contract wraps the script body in an implicit top-level `async` function before executing it — the tool's own canonical examples end with a bare top-level `return <value>` (e.g. `return confirmed`, `return bugs`) for exactly this reason. Biome has no awareness of that DSL and correctly flags a bare `return` as invalid for a plain ES module, but "fixing" it (e.g. wrapping in an explicit IIFE) would break how the harness captures the script's final return value.

I validated this isn't just theoretical — I ran the script live through the `Workflow` tool end-to-end (the classic "frog in a well" puzzle) before pushing: all 8 agents completed, the answer was correct, and both independent verification agents agreed. Leaving the file as-is.

Also noting the "⚔️ Resolve merge conflicts" checkbox in the same review is stale — that reflects the state before my last push (`07954ef`); GitHub's own `mergeable_state` is `unstable` (checks pending), not `dirty`, confirming the conflict is already resolved.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: skills/problem-solving/references/polya-checklist.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+## 計画を実行すること
+
+- 解答の計画を実行するときに、各段階を検討せよ。その段階が正しいことをはっきりと認めることができるか
+
+## 振り返ってみること
+
+- 結果を試すことができるか。議論を試すことができるか
+- 結果を違った仕方で導くことができるか。それを一目のうちに捉えることができるか
+- 他の問題にその結果や方法を応用することができるか
```
[the remark this answers is not in the document]
Comment (kanade0404) [reply skills/problem-solving/references/polya-checklist.md]: Fixed in `94d0712` — replaced the verbatim 柿内賢信-translation reproduction with an original paraphrase (same four sections and guiding questions, fresh wording). No license was on record for the commercial translation, so removing the verbatim text was the right call rather than trying to source permission.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: skills/problem-solving/scripts/four-step-pipeline.workflow.js [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  'pattern-match: has a same or closely similar problem been solved before? Reuse that result/method directly.',
+  'decomposition: split the problem into easier or more specific sub-problems and solve those.',
+  'analogy: find a structurally similar problem and adapt its method.',
+]
+
+const verifyMethods = (resolvedArgs && resolvedArgs.verifyMethods) || [
+  're-derive the result via a genuinely different method and compare',
+  'stress-test the result against an edge case or boundary condition',
+  'check the result actually satisfies the original condition from Step 1',
+]
```
[the remark this answers is not in the document]
Comment (kanade0404) [reply skills/problem-solving/scripts/four-step-pipeline.workflow.js:156]: Fixed in `94d0712` — added an `assertNonEmptyStringArray` check for `planAngles`/`verifyMethods` and a stricter `typeof problem !== 'string' || problem.trim().length === 0` check, all run right after args normalization and before any pipeline work starts.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: skills/problem-solving/scripts/four-step-pipeline.workflow.js [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  )
+).filter(Boolean)
+
+const passCount = verifications.filter((v) => v.passes).length
+log(`Step 4 — ${passCount}/${verifications.length} independent verifications passed`)
+
+const lookback = await agent(
+  `Problem: ${problem}\nResult: ${execution.result}\nIndependent verifications: ${JSON.stringify(verifications)}\n\n${passCount}/${verifications.length} independent verifications passed. Synthesize the Look Back: is the result verified overall? Summarize the strongest confirming and disconfirming evidence, and note what other problems this result or method could generalize to.`,
+  { schema: LOOKBACK_SCHEMA, phase: 'Look Back' }
+)
```
[the remark this answers is not in the document]
Comment (kanade0404) [reply skills/problem-solving/scripts/four-step-pipeline.workflow.js:223]: Fixed in `94d0712` — `lookback.verified` is now overridden in code as `verifications.length > 0 && verifications.every(v => v.passes === true)`, regardless of what the synthesis agent itself claims.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: skills/problem-solving/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+- **Step 4 (振り返り)**: 複数の独立した検証方法 (`verifyMethods`) で並列に結果を検算し、その全結果を踏まえて 1 つの結論に統合する（「別の方法で確かめられるか」を、同一エージェントの自己採点ではなく別エージェントでの再検証にする）
+
+呼び出し方:
+
+```
+Workflow({
+  scriptPath: "skills/problem-solving/scripts/four-step-pipeline.workflow.js",
+  args: { problem: "<完全に条件が確定した問題文>" }  // planAngles / verifyMethods は省略可（既定値あり）
+})
+```
```
[the remark this answers is not in the document]
Comment (kanade0404) [reply skills/problem-solving/SKILL.md]: Fixed in `94d0712` — added the `javascript` language tag to the fence.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by kanade0404** [comment]
## Review Response Summary

未 resolve 0 件

| Reviewer | Total | Fixed | Pushback | Deferred | Duplicate |
|---|---|---|---|---|---|
| CodeRabbit | 4 | 4 | 0 | 0 | 0 |

### Fixed (commit `94d0712`)
- [skills/problem-solving/references/polya-checklist.md#L32](https://github.com/kanade0404/skills/pull/105#discussion_r3742406410) — replaced verbatim 柿内賢信-translation reproduction with an original paraphrase (no license was on record for the commercial translation)
- [skills/problem-solving/scripts/four-step-pipeline.workflow.js#L145](https://github.com/kanade0404/skills/pull/105#discussion_r3742406414) — validate `problem`/`planAngles`/`verifyMethods` right after args normalization, reject invalid shapes with a clear error
- [skills/problem-solving/scripts/four-step-pipeline.workflow.js#L203](https://github.com/kanade0404/skills/pull/105#discussion_r3742406416) — `lookback.verified` is now derived from the actual `verifications` array in code, not trusted from the agent's self-report
- [skills/problem-solving/SKILL.md#L92](https://github.com/kanade0404/skills/pull/105#discussion_r3742406420) — added `javascript` language tag to the Workflow example code fence (MD040)

### Verification
- `node scripts/rulesync-sync.mjs --check`: PASS
- `uv run python3 -m unittest discover -s tests`: PASS (106 tests)
- `uv run python .github/scripts/check_trigger_evals.py`: PASS (exit 0)
- Manual syntax check of the edited `four-step-pipeline.workflow.js` (module body outside `export const meta`): no errors
- Re-ran the pipeline live end-to-end via the `Workflow` tool (same frog-in-a-well test problem as the pre-review validation) to confirm the `lookback.verified` gating and arg-validation changes didn't break real execution

All 4 threads resolved.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by kanade0404** [comment]
Re: the two "outside diff range" findings on `skills/rulesync-sync/SKILL.md` (a file this PR touched during the earlier merge-conflict resolution, not one I originally authored) —

Both VALID, fixed in `4fdd9ef`:
- **Security**: `Bash(uv run*)` in `allowed-tools` would auto-approve any `uv run ...` invocation, including `uv run bash -c ...`. Narrowed to the exact two commands the skill body documents (`uv run python3 -m unittest discover -s tests` and `uv run python .github/scripts/check_trigger_evals.py`).
- **markdownlint**: added the `text` language tag to the skill-directory-layout fence (MD040) and blank lines around both Bash fences in the 生成→検証ワークフロー section (MD031).

Regenerated mirrors via `node scripts/rulesync-sync.mjs`, verified `--check` is clean, and re-ran the full test suite (106 tests) + trigger-evals CI script — all pass.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: skills/problem-solving/scripts/four-step-pipeline.workflow.js [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+  'analogy: find a structurally similar problem and adapt its method.',
+]
+assertNonEmptyStringArray(planAngles, 'planAngles')
+
+const verifyMethods = (resolvedArgs && resolvedArgs.verifyMethods) || [
+  're-derive the result via a genuinely different method and compare',
+  'stress-test the result against an edge case or boundary condition',
+  'check the result actually satisfies the original condition from Step 1',
+]
+assertNonEmptyStringArray(verifyMethods, 'verifyMethods')
```
[the remark this answers is not in the document]
Comment (kanade0404) [reply skills/problem-solving/scripts/four-step-pipeline.workflow.js]: Fixed in `eb113c7` — the default is now used only when the key is strictly `undefined`. Any explicitly-passed value (including `null`/`false`/`""`) now goes through `assertNonEmptyStringArray` unchanged and fails loudly if invalid, per this repo's fail-closed convention. Good catch — the original `||` fallback was exactly the kind of silent-fallback bug that convention exists to prevent.

---
_Generated by [Claude Code](https://claude.ai/code)_
